src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexStatementCondition.java

```diff
@@ -144,13 +144,32 @@ public Condition condition() {
      */
     @Override
     public boolean equals(final Object object) {
-        if (this == object)
+        if (this == object) {
             return true;
-        if (object == null)
+        }
+        if (object == null) {
             return false;
-        if (object.getClass() != this.getClass())
+        }
+        if (object.getClass() != this.getClass()) {
             return false;
+        }
         final IndexStatementCondition cast = (IndexStatementCondition) object;
         return this.value.equals(cast.value) && this.config.equals(cast.config) && this.tokenizer == cast.tokenizer; // expects same reference
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        if (value != null) {
+            result = 31 * result + value.hashCode();
```

Comment:
hashcode

Reply:
Solved in commit 4d4e0023426356b05c592147f65933f69ba5dfa3